### ISSUE TITLE
Allow predicate pushdown for group_by expressions 

### ISSUE THREAD

**Issue by datenzauberai** [pola-rs/polars#11678]
### Description

I found that the predicate pushdown will not "push through" an aggregation. This optimization might have some potential for sloppy queries, e.g. 

`scan_parquet(...).group_by("key").count().filter(pl.col("key").eq(42)).collect()`

 runs much slower than 

`scan_parquet(...).filter(pl.col("key").eq(42)).group_by("key").count().collect()`

**Comment by ritchie46** [comment]
Yes, we should push down predicates that only depend on the keys. Predicates that depend on values should be blocked. Thanks for noticing this.
